### PR TITLE
Rianhughes/rpcv06 integration tests

### LINKED ISSUE #484: rpcv06 integration tests
Implement integration tests for  v3 transactions.

### PR DESCRIPTION
**Pull request by rianhughes**
closes https://github.com/NethermindEth/starknet.go/issues/484

### CODE REVIEW

File: rpc/mock_test.go [hunk 1]
```diff
@@ -527,31 +529,22 @@ func mock_starknet_addDeclareTransaction(result interface{}, method string, args
...
+	case DeclareTxnV2, DeclareTxnV3:
+		deadbeefFelt, err := utils.HexToFelt("0x41d1f5206ef58a443e7d3d1ca073171ec25fa75313394318fc83a074a6631c3")
+		if err != nil {
+			return err
+		}
+		output := AddDeclareTransactionOutput{
+			TransactionHash: deadbeefFelt,
+			ClassHash:       deadbeefFelt,
+		}
+		outputContent, _ := json.Marshal(output)
```
Comment (joshklop) [inline rpc/mock_test.go]: Should this error be checked? (same for a few other places)

File: rpc/mock_test.go [hunk 2]
```diff
@@ -665,28 +642,66 @@ func mock_starknet_addInvokeTransaction(result interface{}, method string, args
...
 		return errors.Wrap(errWrongArgs, fmt.Sprint("wrong number of args ", len(args)))
 	}
-	invokeTx, ok := args[0].(InvokeTxnV1)
+	switch invokeTx := args[0].(type) {
+	case InvokeTxnV1:
+		if invokeTx.SenderAddress != nil {
+			if invokeTx.SenderAddress.Equal(new(felt.Felt).SetUint64(123)) {
+				unexpErr := *ErrUnexpectedError
+				unexpErr.data = "Something crazy happened"
+				return &unexpErr
```
Comment (joshklop) [inline rpc/mock_test.go:690]: So whenever we have a test that has a sender address of `123`, we're testing an unexpected error?

File: rpc/write_test.go [hunk 3]
```diff
@@ -11,82 +12,91 @@ import (
...
-		dec, err := testConfig.provider.AddDeclareTransaction(context.Background(), declareTx)
+		resp, err := testConfig.provider.AddDeclareTransaction(context.Background(), test.DeclareTx)
 		if err != nil {
 			require.Equal(t, err.Error(), test.ExpectedError)
-			continue
-		}
-		if dec.TransactionHash != test.TransactionHash {
-			t.Fatalf("classHash does not match expected, current: %s", dec.ClassHash)
+		} else {
+			require.Equal(t, (*resp.TransactionHash).String(), (*test.ExpectedResp.TransactionHash).String())
```
Comment (joshklop) [inline rpc/write_test.go:88]: Does `test.ExpectedResp.TransactionHash.String()` not work?

File: rpc/mock_test.go (hunk 2 above)
Comment (rianhughes) [reply rpc/mock_test.go:690]: Yes

File: rpc/write_test.go (hunk 3 above)
Comment (rianhughes) [reply rpc/write_test.go:88]: Yes, seems to work (now..)

File: rpc/mock_test.go (hunk 1 above)
Comment (rianhughes) [reply rpc/mock_test.go]: Updated all cases of this in this file

**Review by joshklop — APPROVED** [review]

**Pull request merged**